### ISSUE TITLE
Sticky windows not handled correctly.

### ISSUE THREAD

**Issue by flexibeast** [flexibeast/ewmctrl#3]
As [noted](https://github.com/flexibeast/ewmctrl/issues/2#issuecomment-70111562) by @sanel, `wmctrl` output of -1 to indicate a sticky window is not handled correctly; such windows should not be listed by default.

**Comment by sanel** [comment]
Hm... take a look at my picture: http://imgur.com/JFfnS9O

Looks like here it doesn't dump out -1,  but unsigned int value. Also, I'm not sure is it me or there is a wmctrl bug, but if I hit `wmctrl -lpG` and changed desktop number, open close window and return back, I'm getting sometimes the old window list, but sometimes the new window list. Tried both pekwm and xfwm4 and the behavior is similar.

What wmctrl version are you using?

**Comment by flexibeast** [comment]
i can't get a version number out of `wmctrl`, but the man page for it is dated 12 December 2004.

From inside an XFCE4 session, i've run `wmctrl -lpG` in a terminal whilst using `xfwm4` and then whilst using `pekwm`. `xfce4` lists the sticky windows, with desktop number -1; `pekwm` doesn't list them at all. Do you observe the same behaviour on your system?

i'm sorry, but i don't understand what you're describing in your "is it me or there is a wmctrl bug" sentence; could you rephrase, please?

**Comment by sanel** [comment]
> i can't get a version number out of wmctrl, but the man page for it is dated 12 December 2004.

You probably can check the version of package you have installed. You are on ubuntu/fedora/else?

> i'm sorry, but i don't understand what you're describing in your "is it me or there is a wmctrl bug" sentence; could you rephrase, please?

Sure. Here I presumed I'm having buggy wmctrl version. And indeed it is buggy in my case.

Can you reproduce this case please: in terminal run `wmctrl -lpG`, write down the list of windows. Now open random application (gvim, claws-mail or whatever), and re-run `wmctrl -lpG`. Is window list updated? I'm my case, I'm getting the same list.

Also, regarding to why `pekwm` does not list sticky windows is because plain `pekwm` does not have them; there are no desktop, taskbar or whatever, unlike `xfce4` or other desktop. Can you try to list the windows under kde, if you have it installed?

I presume something is fishy with `wmctrl`, but I can be wrong.

**Comment by flexibeast** [comment]
i'm on Debian Wheezy x86_64. `aptitude versions wmctrl` reports `wmctrl` with a version number of 1.07-7. Keep in mind, though, that since `wmctrl` has no e.g. `-v` or `--version` switch, nor any versioning information in its man page, that "1.07-7" might be the version of the Debian _package_, such that other distros use a different versioning system for their own `wmctrl` package.

> Can you reproduce this case please

i'm happy to,but under which DE? What i've been doing is running an XFCE4 session, first with `xfwm4`, then - _within the same XFCE4 session_ - with `pekwm` via `pekwm --replace`. And that's the context in which sticky windows - such as `xfce4-panel` - are listed by `wmctrl -lpG` when running `xfwm4` but not when running `pekwm`, even though those sticky windows are indeed part of the current X session. Further, in the context of an XFCE4 session with `pekwm`, re-running `wmctrl -lpG` each time i start a new app with a new window results in the new window being listed.

So would you like me to install KDE, use `pekwm` as the WM, and see if `wmctrl -lpG` output changes upon new windows being created? (i'm not using any DE, myself.)

**Comment by sanel** [comment]
Yes, that will be helpful. Strange indeed...

**Comment by flexibeast** [comment]
Okay, so i've just installed `kde-plasma-desktop` 4.8.4. i logged in to the KDE desktop, then ran `pekwm --replace` from a terminal. Having done that, i ran `wmctrl -lpG` - it listed sticky windows, with a desktop number of -1. i then opened more windows (e.g. the Calculator application) - re-running `wmctrl -lpG` correctly listed them. i've also moved windows between workspaces, and the output of `wmctrl -lpG` correctly reflects the changes.

So i'm afraid i'm unable to reproduce the issues you've described, sorry! Sounds like something specific to your setup?

**Comment by sanel** [comment]
Thanks!

I'll presume then it is up to my setup indeed. I will investigate this further, but I think the goal of this ticket is already achieved. You can close it if you would like.

**Comment by flexibeast** [comment]
Okay, thanks! But if you manage to work out what's going on with your system, please do feel free to add an explanation in this thread, as it might later turn out to be useful info for others. :-)

**Comment by sanel** [comment]
Yup, will try to do that if I find the solution
